pagerduty/resource_pagerduty_event_orchestration_integration.go

```diff
@@ -0,0 +1,309 @@
+			AttributePath: p,
+		})
+
+		return diags
+	}
+}
+
+func getEventOrchestrationIntegrationPayloadData(d *schema.ResourceData) (string, *pagerduty.EventOrchestrationIntegration) {
+	orchestrationId := d.Get("event_orchestration").(string)
+
+	integration := &pagerduty.EventOrchestrationIntegration{
+		Label: d.Get("label").(string),
+	}
+
+	return orchestrationId, integration
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oid, payload := getEventOrchestrationIntegrationPayloadData(d)
+
+	retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Creating Integration '%s' for PagerDuty Event Orchestration '%s'", payload.Label, oid)
+
+		if integration, _, err := client.EventOrchestrationIntegrations.CreateContext(ctx, oid, payload); err != nil {
+			if isErrCode(err, 400) {
+				return resource.NonRetryableError(err)
+			}
+			return resource.RetryableError(err)
+		} else if integration != nil {
+			// Try reading an integration after creation, retry if not found:
+			if _, _, readErr := client.EventOrchestrationIntegrations.GetContext(ctx, oid, integration.ID); readErr != nil {
+				log.Printf("[WARN] Cannot locate Integration '%s' on PagerDuty Event Orchestration '%s'. Retrying creation...", integration.ID, oid)
+				return resource.RetryableError(readErr)
+			} else {
+				d.SetId(integration.ID)
+				setEventOrchestrationIntegrationProps(d, integration)
+			}
+		}
+		return nil
+	})
+
+	if retryErr != nil {
+		time.Sleep(2 * time.Second)
+		return diag.FromErr(retryErr)
+	}
+
+	return nil
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := d.Id()
+	oid := d.Get("event_orchestration").(string)
+
+	retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Reading Integration '%s' for PagerDuty Event Orchestration: %s", id, oid)
+
+		if integration, _, err := client.EventOrchestrationIntegrations.GetContext(ctx, oid, id); err != nil {
+			time.Sleep(2 * time.Second)
+			return resource.RetryableError(err)
+		} else if integration != nil {
+			setEventOrchestrationIntegrationProps(d, integration)
+		}
+		return nil
+	})
+
+	if retryErr != nil {
+		return diag.FromErr(retryErr)
+	}
+
+	return nil
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
If the `event_orchestration` has changed we make a call to the `/migration` endpoint to migrate the integration from one EO to another.

If the label attribute has changed we make an Update Integration call to update the label either in the source or the destination orchestration.